### ISSUE TITLE
jline3 doesn't behave properly under cygwin: interprets 'enter' as two new lines

### ISSUE THREAD

**Issue by adorokhine** [jline/jline3#151]
Here is what I experience when typing 'ls' and pressing 'enter' under cygwin:
[2017-07-21 02:01:36] INFO [main] Parsed command line: [ls]
[2017-07-21 02:01:36] INFO [main] Parsed command line: []

Perhaps it's interpreting the extra newline somehow?

**Comment by gnodet** [comment]
What's the output of ```stty -a``` in your terminal ?

**Comment by gnodet** [comment]
Can not reproduce. Please reopen if needed.

**Comment by adorokhine** [comment]
Sorry for the delay. The output is:
```
$ stty -a                                                                                                                                                                                                      
speed 38400 baud; rows 56; columns 207; line = 0;                                                                                                                                                              
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = ^Z; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W; lnext = ^V; flush = ^O; min = 1; time = 0;        
-parenb -parodd cs8 -hupcl -cstopb cread -clocal -crtscts                                                                                                                                                      
-ignbrk brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl ixon -ixoff -iuclc -ixany -imaxbel                                                                                                           
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0                                                                                                                                
isig icanon iexten echo -echoe -echok -echonl -noflsh -tostop -echoctl -echoke
```

**Comment by wdelgenio** [comment]
I'm also running into this issue
`$ stty -a
speed 38400 baud; rows 76; columns 164; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = ^Z; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W;
lnext = ^V; discard = ^O; min = 1; time = 0;
-parenb -parodd cs8 -hupcl -cstopb cread -clocal -crtscts
-ignbrk brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl ixon -ixoff -iuclc ixany imaxbel iutf8
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0
isig icanon iexten echo echoe echok -echonl -noflsh -tostop echoctl echoke -flusho
`

**Comment by gnodet** [comment]
@wdelgenio  Could you give us a way to reproduce the problem ?

**Comment by gnodet** [comment]
Fixed by #520
